graph/src/components/ethereum/types.rs

```diff
@@ -276,6 +277,19 @@ impl EthereumBlockPointer {
     pub fn hash_hex(&self) -> String {
         format!("{:x}", self.hash)
     }
+
+    pub fn earliest_block(blocks: Vec<Self>) -> Option<Self> {
+        blocks
+            .iter()
+            .fold(None, |a: Option<EthereumBlockPointer>, &b| match a {
+                Some(a) => match PartialOrd::partial_cmp(&a.number, &b.number) {
+                    None => None,
+                    Some(Ordering::Less) => Some(a),
+                    Some(_) => Some(b),
+                },
+                None => Some(b),
+            })
+    }
```

Comment:
Would it be possible to use [min_by_key](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.min_by_key) here?